client/apps/edit2/components/content/section_controls/index.jsx

```diff
@@ -34,143 +46,122 @@ export class SectionControls extends Component {
-      return this.props.disabledAlert()
-    }
-    if (this.props.section.get('type') === 'image_set') {
-      this.props.section.set('type', 'image_collection')
-      this.forceUpdate()
-    }
-    this.props.section.set({ layout })
-    this.props.onChange && this.props.onChange()
-  }
+  isScrollingOver = ($section) => {
+    if (this.controls) {
+      const controlsHeight = $(this.controls).height()
+      const offsetTop = $section.offset().top
 
-  toggleImageSet = () => {
-    if (this.props.section.get('type') === 'image_collection') {
-      this.props.section.set({type: 'image_set', layout: 'mini'})
-      this.forceUpdate()
+      const scrollPlusHeader = window.scrollY + this.getHeaderHeight()
+      const offsetMinusControls = offsetTop - controlsHeight
+
+      return scrollPlusHeader > offsetMinusControls
     }
-    this.props.onChange && this.props.onChange()
   }
 
-  hasImageSet () {
-    const { type } = this.props.channel
-    return this.sectionIsImage() && type === 'editorial' || type === 'team'
-  }
+  isScrolledPast = ($section) => {
+    if (this.controls) {
+      const controlsHeight = $(this.controls).height()
+      const sectionHeight = $section.height()
+      const offsetTop = $section.offset().top
+      const scrollPosition = window.scrollY
 
-  sectionIsImage () {
-    return this.props.section.get('type').includes('image')
-  }
+      const scrollPlusControls = scrollPosition + controlsHeight
+      const scrollPlusSection = offsetTop + sectionHeight
 
-  sectionHasFullscreen (section) {
-    const hasFullscreen = ['embed', 'video'].includes(section.get('type')) || this.sectionIsImage()
-    return this.props.articleLayout === 'feature' && hasFullscreen
+      return scrollPlusControls > scrollPlusSection
+    }
   }
 
-  renderSectionLayouts (section) {
-    return (
-      <nav className='edit-controls__layout'>
-        <a
-          name='overflow_fillwidth'
-          className='layout'
-          onClick={() => this.changeLayout('overflow_fillwidth')}
-          data-active={section.get('layout') === 'overflow_fillwidth'} />
-        <a
-          name='column_width'
-          className='layout'
-          onClick={() => this.changeLayout('column_width')}
-          data-active={section.get('layout') === 'column_width'} />
-        {
-          this.sectionHasFullscreen(section) &&
-          <a
-            name='fillwidth'
-            className='layout'
-            onClick={() => this.changeLayout('fillwidth')}
-            data-active={section.get('layout') === 'fillwidth'}>
-            <IconImageFullscreen fill={'white'} />
-          </a>
-        }
-        {
-          this.hasImageSet() &&
-          <a
-            name='image_set'
-            className='layout'
-            onClick={this.toggleImageSet}
-            data-active={section.get('type') === 'image_set'} />
+  insideComponent = () => {
+    const { isHero } = this.props
+    if (this.controls) {
```

Comment:
Minor, but is there a way to exit early from this without the need for a guard? I noticed that in a few spots this is blocking execution. Would it be possible to move it up a bit, outside of this component and into the parent? Something like 

```javascript
{showControls && 
  <SectionControls /> }
```

Reply:
My understanding of the new ref style is that the component doesn't have access to the ref until after mount -- so these are checking to see if it exists before attempting to performing jquery actions on the dom.

Reply:
I see 👍